addons/payment/models/payment_acquirer.py

```diff
@@ -334,7 +355,28 @@ def _get_compatible_acquirers(
         if force_tokenization or self._is_tokenization_required(**kwargs):
             domain = expression.AND([domain, [('allow_tokenization', '=', True)]])
 
+        # Handle currency and amount, convert the amount to the company currency.
+        currency = self.env['res.currency'].browse(currency_id).exists()
+        if currency and amount:
+            company = self.env['res.company'].browse(company_id)
+            date = fields.Date.context_today(self)
+            converted_amount = currency._convert(amount, company.currency_id, company, date)
+
+            domain = expression.AND([
+                domain,
+                ['|', ('maximal_amount', '>=', converted_amount),
+                 '|', ('maximal_amount', '=', False),
+                 ('maximal_amount', '=', 0.00)
+                 ]
+            ])
```

Comment:
I think it will be clearer that way. 
```suggestion
            domain = expression.AND([
                 domain, [ '|', 
                    ('maximal_amount', '>=', converted_amount), 
                    '|', ('maximal_amount', '=', False), ('maximal_amount', '=', 0.00)
                 ]
            ])
```

Reply:
done :)